### PR TITLE
Pick color in verteces

### PR DESCRIPTION
**Pull request by dy**
Add vertex mode of picking color. Really not sure about the option name, if you have better idea than `vertexColor` I’ll rename it.

### CODE REVIEW

File: example/example.js [hunk 1]
```diff
@@ -62,15 +62,16 @@ shell.on("gl-init", function() {
...
   surface = createSurface({
     gl: gl,
     field: field,
     //levels: [contourLevels,contourLevels,contourLevels],
     lineWidth: 3,
     contourTint: 1,
     coords: coords,
-    contourProject: [[true,false,false], [true,false,false], [true,false,false]]
+    contourProject: [[true,false,false], [true,false,false], [true,false,false]],
+    vertexColor: false
```
Comment (etpinard) [inline example/example.js:74]: So, if I'm understanding correctly, in plotly.js we would set `vertexColor` to `true` whenever 

``` js
colorscale[0][0] === colorscale[colorscale.length-1][0]
```

in the surface convert [step](https://github.com/plotly/plotly.js/blob/920a333283b1ad9506e804a2e872eaa7f77c3f2e/src/traces/surface/convert.js#L232) ?

cc @rreusser

**Comment by mikolalysenko** [comment]
Looks ok to me.

**Comment by etpinard** [comment]
What I'm seeing in plotly.js:

image

**Comment by etpinard** [comment]
Oh I fixed it:

image

`vertexColor` must be set currently. 

@dfcreative can you make it such that no `vertexColor` in the params options lead to `vertexColor = 0` ?

**Comment by etpinard** [comment]
@dfcreative 

Nicely done. Your patch is backward compatible once https://github.com/gl-vis/gl-surface3d/pull/8#issuecomment-247720001 is addressed:

image

**Comment by dy** [comment]
@etpinard oh, sorry about that, fixed

**Comment by etpinard** [comment]
Great. It works. Let merge this in.

File: example/example.js (hunk 1 above)
Comment (rreusser) [reply example/example.js:74]: I think that's a good idea, potentially with an obscure manual override.

**Pull request merged**